StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift

```diff
@@ -364,10 +374,21 @@ final class LinkAccountPickerViewController: UIViewController {
                 footerView?.showLoadingView(true)
                 dataSource
                     .markConsentAcquired()
-                    .observe { [weak self] _ in
+                    .observe { [weak self] result in
                         guard let self = self else { return }
                         footerView?.showLoadingView(false)
-                        pushToNextPane()
+                        switch result {
+                        case .success:
+                            pushToNextPane()
+                        case .failure(let error):
+                            self.dataSource.analyticsClient.logUnexpectedError(
+                                error,
+                                errorName: "ConsentAcquiredError",
```

Comment:
Do you think we should we make this error name a little more granular to distinguish this from the above one?

Reply:
yea I was thinking about that too....there's definitely a _chance_ it matters to differentiate, but I can't necessarily come up with a good reason why it would really matter so I'm going with the "simple first step" (make them both the same) [side-note on why not differentiate?: one can make an argument that if the errors are essentially the same, having them be different can add to confusion - too much extra analysis when one shouldn't even begin that analysis]